packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -444,21 +539,23 @@ class _SnackBarState extends State<SnackBar> {
     final SnackBarThemeData snackBarTheme = theme.snackBarTheme;
     final bool isThemeDark = theme.brightness == Brightness.dark;
     final Color buttonColor = isThemeDark ? colorScheme.primary : colorScheme.secondary;
+    final SnackBarThemeData defaults = theme.useMaterial3
+        ? _SnackbarDefaultsM3(context)
+        : _SnackbarDefaultsM2(context);
 
     // SnackBar uses a theme that is the opposite brightness from
     // the surrounding theme.
     final Brightness brightness = isThemeDark ? Brightness.light : Brightness.dark;
-    final Color themeBackgroundColor = isThemeDark
-      ? colorScheme.onSurface
-      : Color.alphaBlend(colorScheme.onSurface.withOpacity(0.80), colorScheme.surface);
+
+    // Invert the theme values for Material 2. Material 3 values are tokenzied to pre-inverted values.
```

Comment:
Probably not critical, but we shouldn't create the inverse theme if we don't need it. Perhaps something like:

```dart
final ThemeData effectiveTheme = theme.useMaterial3
  ? theme
  : theme.copyWith(
      colorScheme: // ... inverse color scheme for M2.
```

And then use `effectiveTheme` below when building the Material widget.

Reply:
Good comment, thank you